data-solutions/data-platform-foundations/02-resources/main.tf

```diff
@@ -25,21 +25,30 @@ module "datamart-sa" {
   iam_project_roles = {
     "${var.project_ids.datamart}" = ["roles/editor"]
   }
+  iam = var.admins != null ?  { "roles/iam.serviceAccountTokenCreator" =  concat(var.admins) } : {}
 }
 
 module "dwh-sa" {
   source     = "../../../modules/iam-service-account"
   project_id = var.project_ids.dwh
   name       = var.service_account_names.dwh
+  
+   iam_project_roles = {
```

Comment:
I think the linter will fail here. Make sure to run all terraform files through `terarform fmt`

Reply:
:+1: you can set vscode to format on save, and forget about formatting in the future

Reply:
fixed